### PR TITLE
Add managed hub policy set and placement

### PR DESCRIPTION
**Pull request by sahare**
# Description

- Adding a new policy set containing the existing _backup-restore-enabled_ policy for managed hubs. A new placement rule is added for managed hubs which is targeting clusters with label is-hub=true.
- Reverting placement back to placement rule for local-cluster due to the issue explained below:

The issue occurred with this change https://github.com/stolostron/multiclusterhub-operator/pull/1289 replacing placement rule with placement. The problem is if local-cluster is not part of the default cluster set, the required ManagedClusterSetBinding doesn't exist in the open-cluster-management-backup namespace and so the policy is not placed.

## Related Issue

https://issues.redhat.com/browse/ACM-9629

## Changes Made

- New policy set: acm-managed-hub-backup
- New placement rule: acm-managed-hub-backup
- New placement binding: acm-managed-hub-backup
- Reverting placement acm-hub-backup back to placement rule and targeting clusters named local-cluster

## Screenshots (if applicable)

Add screenshots or GIFs that demonstrate the changes visually, if relevant.

## Checklist

- [x] I have tested the changes locally and they are functioning as expected.
- [ ] I have updated the documentation (if necessary) to reflect the changes.
- [ ] I have added/updated relevant unit tests (if applicable).
- [x] I have ensured that my code follows the project's coding standards.
- [x] I have checked for any potential security issues and addressed them.
- [ ] I have added necessary comments to the code, especially in complex or unclear sections.
- [x] I have rebased my branch on top of the latest main/master branch.

## Additional Notes

Add any additional notes, context, or information that might be helpful for reviewers.

## Reviewers

Tag the appropriate reviewers who should review this pull request. To add reviewers, please add the following line: `/cc @reviewer1 @reviewer2`

## Definition of Done

- [ ] Code is reviewed.
- [ ] Code is tested.
- [ ] Documentation is updated.
- [ ] All checks and tests pass.
- [ ] Approved by at least one reviewer.
- [ ] Merged into the main/master branch.

### CODE REVIEW

File: pkg/templates/charts/toggle/cluster-backup/templates/managed-hub-placementrule.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+  annotations:
+    "helm.sh/hook": pre-install,post-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation
+  name: acm-managed-hub-backup
+  namespace: open-cluster-management-backup
+spec:
+  clusterSelector:
```
Comment (birsanv) [inline pkg/templates/charts/toggle/cluster-backup/templates/managed-hub-placementrule.yaml:10]: should we have here this as well ?

  clusterConditions:
    - status: 'True'
      type: ManagedClusterConditionAvailable

Comment (sahare) [reply pkg/templates/charts/toggle/cluster-backup/templates/managed-hub-placementrule.yaml:10]: This condition was removed in your previous PR and I don't have a managed hub to see if we really need it. I tested with a is-hub=true label only and the placement works. If you think we need it I can add it here too.

Comment (birsanv) [reply pkg/templates/charts/toggle/cluster-backup/templates/managed-hub-placementrule.yaml:10]: this is to match what is now with the hub placement - try to place only on managed clusters which are in active state
I am not really sure what is the implication if we don't have this here ( if the managed cluster is not available, the hub can't connect to it anyway ). If we have this for the local-cluster , let's add it to the managed clusters too

**Review by dislbenn — APPROVED** [review]

**Pull request merged**